compiler-plugin/src/main/kotlin/arrow/meta/quotes/Quote.kt

```diff
@@ -318,14 +313,32 @@ inline fun <reified K : KtElement> Transform.Many<K>.many(ktFile: KtFile, compil
 }
 
 fun <K : KtElement> Transform.Replace<K>.replace(file: Node.File, context: PsiElement? = null): Node.File = MutableVisitor.preVisit(file) { element, _ ->
-    if (element != null && element == (context?.ast ?: replacing.ast)) {
-        val newContents = newDeclarations.joinToString("\n") { it.value?.text ?: "" }
+    if (element != null && element == getAst(context, replacing)) {
+        val newContents = getNewContent(replacing, newDeclarations)
         println("Replacing ${element.javaClass} with ${newDeclarations.map { it.value?.javaClass }}: newContents: \n$newContents")
         element.dynamic = newContents
         element
     } else element
 }
 
+fun getNewContent(replacing: PsiElement, newDeclarations: List<Scope<KtElement>>): String? {
+  return if (replacing is KtPropertyAccessor) {
+    newDeclarations.joinToString("\n") { (it.value as KtPropertyAccessor).property?.text ?: "" }
+  } else {
+    newDeclarations.joinToString("\n") {
+      it.value?.text ?: "" }
+  }
+}
+
+
+fun getAst(context: PsiElement?, replacing: PsiElement): Node {
```

Comment:
The same question about move this logic to other place here.

Reply:
yes! thanks!